### PR TITLE
add post draft

### PR DESCRIPTION
**Pull request by ciaranschutte**

### CODE REVIEW

File: _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
+---
+layout: post
```
Comment (rosibaj) [inline _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md:2]: section needs to be fixed.

File: _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+title: "YOUR BLOG TITLE"
```
Comment (rosibaj) [inline _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md]: title? teaser fix?

File: _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+CSS-in-JS. A phrase which splits opinion in two nearly as much as how delicious cilantro is.
+
+I will present a grab bag of opinions on why you might want to consider using it and share with you some reasons why I like using it.
+At the end I will make note of some recent reflections.
+
+This article mostly considers React web apps using Emotion 10.
+
+## CSS-in-JS
+
+CSS-in-JS is just that. Instead of writing rule ridden CSS stylesheets we now write javascript functions which return us styled components. CSS modules allowed us to co-locate our styles with our components but now we can go a step further and write our styles completely inline with our components. This benefits code organisation. We are also scoping our styles to our components, instead of polluting a global scope. If we delete our component, the styles go to. No more useless styles hanging out after midnight and bloating our precious bundle size for first render.
```
Comment (rosibaj) [inline _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md]: suggested syntax for grammar:  
...If we delete our component, the styles `are also removed`.

File: _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+---
+
+## Introduction
+
+CSS-in-JS. A phrase which splits opinion in two nearly as much as how delicious cilantro is.
+
+I will present a grab bag of opinions on why you might want to consider using it and share with you some reasons why I like using it.
+At the end I will make note of some recent reflections.
+
+This article mostly considers React web apps using Emotion 10.
```
Comment (rosibaj) [inline _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md]: link emotion: [Emotion 10](https://emotion.sh/docs/introduction).

File: _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+
+Using this approach we're constrained to this awful halfway mark between an object and CSS properties. Remembering to camel case properties and use strings for number values when our head is in CSS world constantly trips me up. CSS-in-JS provides a more natural inline style for us and it even comes with syntax highlighting in VSCode!
+
+```
+<h1 css={css`font-weight: bold; font-size: 22px`}>2nd Heading</h1>
+```
+
+## Extending
+
+This works great for small styles or specific overrides but what if we have a lot of styling? For example a large button component. With Emotion we can created a styled component like so:
```
Comment (rosibaj) [inline _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md:65]: small grammar edits: 
```This works great for small styles or specific overrides, but what if you have a lot of styling? For example, a large button component. With Emotion we can created a styled component like so:
```

File: _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+## Props to Emotion
+
+All approaches using Emotion can incorporate our components props in anyway we see fit. After all it's JavaScript. There is no need to toggle classnames for interaction styles, or any other dynamic styling you may want. Simply use props:
+
+```
+const MenuItem = stlyed(Item)`
+  color: ${({active}) => active ? 'blue' : 'white'}`
+```
+
+Our MenuItem color will blue if it's active, or white if it's not. A contrived example but you can extrapolate on how powerful this flexibility may be.
```
Comment (rosibaj) [inline _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md]: Styleing/grammar edits:

Our `MenuItem` color will blue if it's active, or white if it's not. A contrived example, but you can extrapolate on how powerful this flexibility may be.

File: _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md (hunk 3 above)
Comment (rosibaj) [inline _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md]: Comment for through out - seeing JavaScript cased different ways.  stick to just one (i think JavaScript is the most correct).

File: _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+    color: ${({ theme, variant }) => theme.button.textColors[variant].disabled};
+  }
+`;
+```
+
+You can see from this example how we can change a variable in our theme file and have it affect our button styles.
+
+## Where do we go now?
+
+I've been using Emotion for over 2 years. It's developer experience is top notch, removing a lot of obstacles to just getting code running and a ticket complete.
```
Comment (rosibaj) [inline _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md:216]: Small grammar edits:
....
I've been using Emotion for over 2 years. It's developer experience is top notch, removing a lot of obstacles to just getting code running and a ticket complete.
This is`, however,` a double edged sword. I enjoy not thinking about classnames and writing styles inline but I've also found myself not reusing a lot of styles.

Using a library like this brings with it tooling considerations. Integrating with Babel, or Webpack`,` or using custom pragma annotations are extra considerations.
....

Comment (samrichca) [inline _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md:216]:
```suggestion
I've been using Emotion for over 2 years. Its developer experience is top notch, removing a lot of obstacles to just getting code running and a ticket complete.
```

File: _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+This is a double edged sword. I enjoy not thinking about classnames and writing styles inline but I've also found myself not reusing a lot of styles.
+
+Using a library like this brings with it tooling considerations. Integrating with Babel, or Webpack or using custom pragma annotations are extra considerations.
+Additionally, in constrast to CSS preprocessor tooling, you're also shipping a library which can impact performance.
```
Comment (samrichca) [inline _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md]:
```suggestion
Additionally, in contrast to CSS preprocessor tooling, you're also shipping a library which can impact performance.
```

File: _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+
+There are aspects of Emotion I don't fully understand which are also worth considering. It's not zero runtime (unlike a library like Linaria) so there will be performance considerations.
+
+Be aware of your team. It's far easier for a designer or anyone else to read some basic CSS than to trawl through intimidating components.
+
+SASS with CSS modules will get you a very long. Stay tuned for more.
```
Comment (samrichca) [inline _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md]:
```suggestion
SASS with CSS modules will get you a very long. Stay tuned for more.
```

missing word?

File: _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+Once you ensure your react component can receive a `className` prop correctly to it's children, you can use `styled` on your component:
+
+```
+const PrettyCoolComponent = ({children, className, ...props}) =>
+  <div className={className}>{children}</div>
+```
+
+## Emotional Rescue
+
+As projects grow larger there is a concern for unsued code and dangling styles.
```
Comment (samrichca) [inline _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md]:
```suggestion
As projects grow larger there is a concern for unused code and dangling styles.
```

File: _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md [hunk 11]
```diff
@@ -0,0 +1,224 @@
...
+
+```
+const FormButton = styled(DefaultButton)`
+  text-transform: none;
+  font-weight: 500;
+  border-raduis: 0
+`;
+```
+
+Once you ensure your react component can receive a `className` prop correctly to it's children, you can use `styled` on your component:
```
Comment (samrichca) [inline _posts/ciaran_schutte/2020-22-10-CSS-in-JS.md]:
```suggestion
Once you ensure your react component can receive a `className` prop correctly to its children, you can use `styled` on your component:
```

**Comment by ciaranschutte** [comment]
added links, fixed metadata, fixed author image, added post image

**Comment by rosibaj** [comment]
@ciaranschutte do you have a netlify now? I'd like to see it in blog now :)

**Review by rosibaj — APPROVED** [review]

**Pull request merged**